### PR TITLE
[TH2-1537] There is no way to change pageSize and timeout parameters in schema version of the cradle-viewer

### PR DESCRIPTION
**Pull request by MrJonniK**

### CODE REVIEW

File: src/main/java/com/exactpro/th2/common/schema/cradle/CradleConfiguration.java [hunk 1]
```diff
@@ -42,6 +42,12 @@
 
     @JsonProperty
     private String cradleInstanceName;
+    
+    @JsonProperty
+    private long timeout;
+    
+    @JsonProperty
+    private int pageSize;
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/common/schema/cradle/CradleConfiguration.java:50]: Please define default values

**Review by Nikita-Smirnov-Exactpro — COMMENTED** [review]
Please bump the minor version

File: src/main/java/com/exactpro/th2/common/schema/cradle/CradleConfiguration.java (hunk 1 above)
Comment (MrJonniK) [reply src/main/java/com/exactpro/th2/common/schema/cradle/CradleConfiguration.java:50]: Default value for instance variables for int and long is 0. When they are set to 0 default value inside cradle api is used. Should I set them explicitly?

Comment (Nikita-Smirnov-Exactpro) [reply src/main/java/com/exactpro/th2/common/schema/cradle/CradleConfiguration.java:50]: Please define explicitly for understanding

File: gradle.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
-release_version=3.2.2
+release_version=3.2.3
```
Comment (Nikita-Smirnov-Exactpro) [inline gradle.properties:1]:
```suggestion
release_version=3.3.0
```

File: README.md [hunk 3]
```diff
@@ -82,7 +84,9 @@ The `CommonFactory` reads a Cradle configuration from the cradle.json file.
   "password": "<password>",
   "cradleInstanceName": "<cradle instance name>",
   "cradleMaxEventBatchSize": 1048576,
-  "cradleMaxMessageBatchSize": 1048576
+  "cradleMaxMessageBatchSize": 1048576,
+  "timeout": 5000,
+  "pageSize": 0
```
Comment (Nikita-Smirnov-Exactpro) [inline README.md:89]:
```suggestion
  "pageSize": 5000
```

File: src/main/java/com/exactpro/th2/common/schema/cradle/CradleConfiguration.java [hunk 4]
```diff
@@ -43,6 +43,18 @@
...
     private String cradleInstanceName;
 
+    /**
+     * This is the connection timeout {@link CassandraStorageSettings#DEFAULT_TIMEOUT} set as the default value.
+     */
+    @JsonProperty
+    private long timeout = CassandraStorageSettings.DEFAULT_TIMEOUT;
+
+    /**
+     * This is the connection timeout {@link CassandraStorageSettings#DEFAULT_TIMEOUT} set as the default value.
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/common/schema/cradle/CradleConfiguration.java:53]: Correct readme

File: src/main/java/com/exactpro/th2/common/schema/cradle/CradleConfiguration.java [hunk 5]
```diff
@@ -43,6 +43,18 @@
...
+     */
+    @JsonProperty
+    private int pageSize = 0;
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/common/schema/cradle/CradleConfiguration.java:56]:
```suggestion
    private int pageSize = 5000;
```
or value from const

File: README.md [hunk 6]
```diff
@@ -71,6 +71,8 @@ The `CommonFactory` reads a Cradle configuration from the cradle.json file.
 * cradleInstanceName - this option defines a special identifier that divides data within one keyspace with infra as the default value.
 * cradleMaxEventBatchSize - this option defines the maximum event batch size in bytes with its default value set to 1048576.
 * cradleMaxMessageBatchSize - this option defines the maximum message batch size in bytes with its default value set to 1048576.
+* timeout - this option defines connection timeout in millisecons. If set to 0 or ommited, the default value 5000 is used.
```
Comment (jean-michel-perli-sonsin) [inline README.md:74]: timeout - this option defines connection timeout in **milliseconds**. If set to 0 or ommited, the default value **of** 5000 is used.

File: README.md [hunk 7]
```diff
@@ -71,6 +71,8 @@ The `CommonFactory` reads a Cradle configuration from the cradle.json file.
...
+* pageSize - this option defines size of result set to fetch at a time. If set to 0 or ommited, the default value from cassandra is used. Usually 5000. It depends on cassandra version.
```
Comment (jean-michel-perli-sonsin) [inline README.md:75]: pageSize - this option defines **the** size of **the** result set to fetch at a time. If **it is** set to 0 or ommited, the default value from cassandra is used. Usually **it is set to** 5000. It **will depend on the version of cassandra**.

File: src/main/java/com/exactpro/th2/common/schema/cradle/CradleConfiguration.java [hunk 8]
```diff
@@ -43,6 +43,18 @@
...
+
     /**
      * TThis is the maximum event batch size in bytes with {@link CassandraStorageSettings#DEFAULT_MAX_EVENT_BATCH_SIZE} set as the default value.
```
Comment (jean-michel-perli-sonsin) [inline src/main/java/com/exactpro/th2/common/schema/cradle/CradleConfiguration.java:59]: **This** is the maximum event batch size in bytes with {@link CassandraStorageSettings#DEFAULT_MAX_EVENT_BATCH_SIZE} set as the default value.

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Review by jean-michel-perli-sonsin — CHANGES_REQUESTED** [review]
Hi, please find below some changes that I made at parts of the text:

README.md
Line 70:
password - the required setting defines the password that will be used for **connecting** to Cassandra

src/main/java/com/exactpro/th2/common/schema/factory/AbstractCommonFactory.java
Line 148:
Create factory with **non default** implementations schema classes
Line 187:
CommonFactoryException if **cannot** call **the** default constructor from class
Line 188:
IllegalStateException  if **cannot** read configuration
Line 197:
CommonFactoryException(" **Cannot** create parsed message router", e);
Line 207:
CommonFactoryException if **cannot** call **the** default constructor from class
Line 208:
IllegalStateException  if **cannot** read configuration
Line 217:
CommonFactoryException("**Cannot** create raw message router", e);
Line 227:
CommonFactoryException if **cannot** call **the** default constructor from class
Line 228:
IllegalStateException  if **cannot** read configuration
Line 237:
CommonFactoryException("**Cannot** create group message router", e);
Line 247:
CommonFactoryException if **cannot** call **the** default constructor from class
Line 248:
IllegalStateException  if **cannot** read configuration
Line 257:
CommonFactoryException("**Cannot** create event batch router", e);
Line 266:
CommonFactoryException if **cannot** call **the** default constructor from class
Line 267:
IllegalStateException  if **cannot** read configuration
Line 276:
CommonFactoryException("**Cannot** create GRPC router", e);
Line 302:
messageClass **-** custom message class
Line 303:
messageConverter - **the** converter that will **be** used to convert message to bytes and vice versa
Line 304:
defaultSendAttributes **- the** set of attributes for sending **messages**. A pin must have all of them **selected** for sending the message
Line 305:
defaultSubscribeAttributes **- the** set of attributes subscription. A pin must have all of them **selected** for receiving messages
Line 332:
It the router for that type is **still not registered yet, it throws** {@link IllegalArgumentException}
Line 335:
IllegalArgumentException if **the router for the type specified** is not registered
Line 336:
the previously registered router for **the type specified**
Line 351:
IllegalStateException if **cannot** read configuration
Line 422:
Parse json file with custom configuration to java bean using custom {@link ObjectMapper} to deserialize **the** file's content.
Line 426:
Java bean with custom configuration, or <b>NULL</b> if **the configuration does not exist** and **cannot call the default** constructor from java bean class
Line 427:
IllegalStateException if **cannot** read configuration
Line 443:
Parse json file with **the** custom configuration to java bean. This method uses **the** default {@link ObjectMapper}.
Line 444:
If you need **a** custom setting for deserialization use {@link #getCustomConfiguration(Class, ObjectMapper)} method.
Line 447:
Java bean with custom configuration, or <b>NULL</b> if **the configuration does not exist** and **cannot call the** default constructor from java bean class
Line 448:
IllegalStateException if **cannot** read configuration
Line 456:
IllegalStateException if **cannot** read dictionary
Line 463:
dictionaryType **- the **desired type of dictionary
Line 465:
IllegalStateException if **cannot** read dictionary

**Pull request merged**